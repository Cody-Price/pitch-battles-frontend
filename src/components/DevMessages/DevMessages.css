.dev-message {
  font-family: "chronotype";
  padding: 1rem;
  position: relative;
  animation: fade-in-dev-message 0.3s linear;
  opacity: 100%;
}

.dev-message-title {
  font-size: 1.5rem;
  margin-top: 1.5rem;
  margin-bottom: 0.5rem;
}

.dev-message p {
  padding: 0 1.5rem;
  font-size: 1.2rem;
}

.dev-message button {
  background: url("../Landing/images/close-btn.png");
  position: absolute;
  top: 2rem;
  right: 2rem;
  border: none;
  width: 20px;
  height: 18px;
  background-size: cover;
  image-rendering: pixelated;
}

.dev-message button:hover {
  cursor: pointer;
}

@keyframes fade-in-dev-message {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 100%;
  }
}
