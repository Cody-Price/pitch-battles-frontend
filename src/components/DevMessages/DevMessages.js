import React, { Component } from "react";
import "./DevMessages.css";

import { devMessages } from "../../content/devMessages";

class DevMessages extends Component {
  render() {
    const content = devMessages.text.map((message, index) => {
      return (
        <p key={`devMessage ${index}`} className="dev-message-text">
          {message}
        </p>
      );
    });

    return (
      <article className="dev-message">
        <button
          className="close-dev-message"
          onClick={this.props.closeDevMessage}
        />
        <h4 className="dev-message-title">{devMessages.title}</h4>
        <p className="dev-message-date">{devMessages.date}</p>
        {content}
        <div />
      </article>
    );
  }
}

export default DevMessages;
