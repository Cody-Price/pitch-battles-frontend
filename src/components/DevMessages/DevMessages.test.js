import React from "react";
import DevMessages from "./DevMessages";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("DevMessages", () => {
  let wrapper;
  let mockCloseDevMessage;

  beforeEach(() => {
    mockCloseDevMessage = jest.fn();
    wrapper = shallow(<DevMessages closeDevMessage={mockCloseDevMessage} />);
  });

  it("should match the snapshop", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call closeDevMessage on click", () => {
    wrapper.find(".close-dev-message").simulate("click");

    expect(mockCloseDevMessage).toHaveBeenCalled();
  });
});
