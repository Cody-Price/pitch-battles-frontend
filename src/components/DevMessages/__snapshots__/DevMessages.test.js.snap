// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevMessages should match the snapshop 1`] = `
<article
  className="dev-message"
>
  <button
    className="close-dev-message"
    onClick={[MockFunction]}
  />
  <h4
    className="dev-message-title"
  >
    Database Reset
  </h4>
  <p
    className="dev-message-date"
  >
    Thursday, January 17th, 2018
  </p>
  <p
    className="dev-message-text"
    key="devMessage 0"
  >
    Now that Pitch Battles is beginning to see some traffic, we have had to rollback and migrate our database to accomodate a growing user population. If you previously created an account, you will need to create a new one. This should be the last time this happens pending some unforeseen catastrophe.
  </p>
  <p
    className="dev-message-text"
    key="devMessage 1"
  >
    Additionally, we have been adding loads of small improvements to the student-facing user interface including support for iOS browsers, but we are still looking for user feedback. If you find any bugs, please send us an email at teampitchbattles@gmail.com!
  </p>
  <div />
</article>
`;
